crates/cubecl-cuda/src/runtime.rs

```diff
@@ -155,8 +162,22 @@ fn create_client<M: DialectWmmaCompiler<CudaDialect<M>>>(
         }
     };
 
-    let memory_management =
-        MemoryManagement::from_configuration(storage, &mem_properties, options.memory_config);
+    let memory_management_gpu = MemoryManagement::from_configuration(
+        storage,
+        &mem_properties,
+        options.memory_config.clone(),
+    );
+    // We use the same page size and memory pools configuration for CPU pinned memory, since we
+    // expect the CPU to have at least the same amount of RAM as GPU memory.
+    let memory_management_cpu = MemoryManagement::from_configuration(
+        PinnedMemoryStorage::new(),
+        &MemoryDeviceProperties {
+            max_page_size: mem_properties.max_page_size,
+            alignment: PINNED_MEMORY_ALIGNMENT as u64,
+            data_transfer_async: false,
+        },
+        options.memory_config,
+    );
```

Comment:
That's not necessarily a good assumption, as evidenced by my 5090 with only 32GB of total system RAM 😅 
That being said, it should only grow as big as needed for the transfer, correct? If so it might be fine.